### PR TITLE
EN-7

### PR DESCRIPTION
**Pull request by marcosihe**
-Refactor forms into CreateForm and EditForm
-Modify views: clients, employees, contracts, company, project, projectTypes, roles, and seniorities, in order to be able to use CreateForm.jsx and EditForm.jsx
-Refactor the data contained in the forms, creating the const formData.

### CODE REVIEW

File: src/projectTypes.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+import { Datagrid, List, TextField, EditButton } from "react-admin";
+import CreateForm from "./components/forms/CreateForm";
+import EditForm from "./components/forms/EditForm";
 
-const inputsList = [
-    {name: "name"}
+const formData = [
+  {
+    title: "Project Type",
+    inputsList: [{ name: "name" }],
```
Comment (agustinb4) [inline src/projectTypes.js:9]: Same here:  `inputsList: [{ name: "name", type:"string" }],`

**Review by agustinb4 — CHANGES_REQUESTED** [review]
Good job man!

File: src/roles.js [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-import { Datagrid, List, TextField, Edit, SimpleForm, TextInput, Create, EditButton } from 'react-admin';
+import * as React from "react";
+import { Datagrid, List, TextField, EditButton } from "react-admin";
+import CreateForm from "./components/forms/CreateForm";
+import EditForm from "./components/forms/EditForm";
+
+const formData = [
+  {
+    title: "Roles",
+    inputsList: [{ name: "name" }],
```
[the remark this answers is not in the document]
Comment (marcosihe) [reply src/roles.js:9]: you're right! thanks for noticing that!

File: src/projectTypes.js (hunk 1 above)
Comment (marcosihe) [reply src/projectTypes.js]: Done

File: src/roles.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcosihe) [reply src/roles.js]: Done

File: src/contracts.js [hunk 3]
```diff
@@ -1,79 +1,104 @@
...
+  },
+  {
+    title: "Employee",
+    inputsList: [
+      { name: "startDate", type: "date" },
+      { name: "endDate", type: "date" },
+    ],
+    referenceValues: {
+      source: "employeeId",
+      reference: "companies",
```
[the remark this answers is not in the document]
Comment (marcosihe) [reply src/contracts.js]: Done, nice catch!

File: src/components/forms/CreateForm.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import FormSection from "./FormSection";
+
+const CreateForm = ({ formData, title }) => {
+  return (
+    <Box m="10px">
+      <Box display="flex" justifyContent="space-between" alignItems="center">
+        <Header title={title} subtitle="Create" />
+      </Box>
+
+      <Create>
```
[the remark this answers is not in the document]
Comment (marcosihe) [reply src/components/forms/CreateForm.jsx]: Done!

File: src/contracts.js [hunk 5]
```diff
@@ -1,79 +1,107 @@
...
+      multiselect: false,
+    },
+  },
+  {
+    title: "Seniority",
+    referenceValues: {
+      source: "seniorityId",
+      reference: "seniorities",
+      optionText: "name",
+      multiselect: true,
```
Comment (agustinb4) [inline src/contracts.js:57]: should be false

File: src/contracts.js [hunk 6]
```diff
@@ -1,79 +1,107 @@
...
+  {
+    title: "Employee",
+    inputsList: [
+      { name: "startDate", type: "date" },
+      { name: "endDate", type: "date" },
+    ],
+    referenceValues: {
+      source: "employeeId",
+      reference: "employees",
+      optionText: "name",
```
Comment (agustinb4) [inline src/contracts.js:33]: should be lastName

File: src/contracts.js [hunk 7]
```diff
@@ -1,79 +1,107 @@
...
+    title: "Seniority",
+    referenceValues: {
+      source: "seniorityId",
+      reference: "seniorities",
+      optionText: "name",
+      multiselect: true,
+    },
+  },
+  {
+    paymentInformation: false,
```
Comment (agustinb4) [inline src/contracts.js:61]: please remove from all component on false cases  (if no necessary)

Comment (marcosihe) [reply src/contracts.js]: Done

File: src/contracts.js (hunk 6 above)
Comment (marcosihe) [reply src/contracts.js]: Done

File: src/contracts.js (hunk 5 above)
Comment (marcosihe) [reply src/contracts.js]: Done

**Review by agustinb4 — APPROVED** [review]
:rocket:

**Pull request merged**